### ISSUE TITLE
Run run rake task only once on next deploy

### ISSUE THREAD

**Issue by rusikf** [capistrano/capistrano#2057]
Hi, is it possible to run custom command( ex: rake task ) **only once on next deploy only?**
I don't wan't to wait deploy, I want to run rake task only once ( On production for compatibility ), without  running it manually and without rails migrations

**Comment by leehambley** [comment]
If this solves your problem, please feel free to close the issue:

- https://stackoverflow.com/a/18840098/119669

**Comment by rusikf** [comment]
But it describes, how to run task manually, Not to Run Once automatically after next deploy only ( like rails migration )

**Comment by leehambley** [comment]
I'm sorry I'm confused, in the first post you asked:

> I don't wan't to wait deploy, I want to run rake task only once ( On production for compatibility ), without running it manually and without rails migrations

Sounds like you *don't* want this coupled to a deploy.

The easy way is to make use of rake task dependencies - https://subscription.packtpub.com/book/hardware_and_creative/9781783280773/1/ch01lvl1sec13/task-dependencies-prerequisites

You can definitely hook in to an existing rake task, or extend the capistrano deploy task with an extra step (capistrano's tasks/steps DSL is literally just rake)

**Comment by rusikf** [comment]
yes, I understand that I can run any rake task on deploy, but it is possible to restrict run it only ONCE ? Looks, only if I add additional logic to redis or save to capistrano shared files, and check it, but may be it is possible with capistrano ? To save data, that rake task already run between deploys

**Comment by leehambley** [comment]
> yes, I understand that I can run any rake task on deploy, but it is possible to restrict run it only ONCE ? 

Delete the code after you ran it? Use a lockfile, or a locking key in Redis?

**Comment by weedySeaDragon** [comment]
I've written a task (and the supporting) to do this.  
You definitely have to have some way to record that you have run the task (e.g. in some sort of persistence: redis, a text file, db, ... whatever).

The code is specific to our system (how we decided to record if a task has been run), but if you still need help or want info about this, let me know.
